context.go

```diff
@@ -32,290 +65,45 @@ func (c *Context) FindVariable(e Expression) (Expression, error) {
 		return nil, fmt.Errorf("DUNNO WHAT TO DO WITH %v OMG", e)
 	}
 
-	if value, ok := c.vars[key]; ok {
+	scope, err := c.scopes.curr()
+	if err != nil {
+		return nil, err
+	}
+
+	if value, ok := scope[key]; ok {
```

Comment:
Small semantic difference: Get() will return `nil` if the variable doesn't exist, whereas the current impl will return an ErrNotFound error.  Since it's technically valid to store and return a nil or operate on a nil it's kind of a nuanced decision what we want to do here.  I opted to leave the current behavior intact incase you'd built anything ontop of it.

I imagine it will be changed pretty soon as the pieces come together, especially error handling, but I don't know exactly what it'll look like.  If you'd rather it changed that's cool, but I'm tempted to lack inertia/need decide when to change it.